File: 2017-multi-jpa-java/core/src/main/java/sample/member/core/transfer/ListResponseWrapper.java
package sample.member.core.transfer;

import lombok.Data;

import java.util.List;

@Data
public class ListResponseWrapper<RESPONSE> {

	public static <RESPONSE0> ListResponseWrapper<RESPONSE0> create() {
		return new ListResponseWrapper<>();
	}

	private boolean success = true;

	public ListResponseWrapper<RESPONSE> success(boolean success) {
		this.success = success;
		return this;
	}

	private String lang;

	public ListResponseWrapper<RESPONSE> lang(String lang) {
		this.lang = lang;
		return this;
	}

	private String code;

	public ListResponseWrapper<RESPONSE> code(String code) {
		this.code = code;
		return this;
	}

	private String message;

	public ListResponseWrapper<RESPONSE> message(String message) {
		this.message = message;
		return this;
	}

	private List<RESPONSE> data;

	public ListResponseWrapper<RESPONSE> data(List<RESPONSE> data) {
		this.data = data;
		return this;
	}
}

File: 2021-rabbitmq-jpa/core/src/main/java/sample/core/transfer/ListResponseWrapper.kt
package sample.core.transfer

class ListResponseWrapper<RESPONSE> {
    private var success = true
    fun success(success: Boolean): ListResponseWrapper<RESPONSE> {
        this.success = success
        return this
    }

    private var lang: String? = null
    fun lang(lang: String?): ListResponseWrapper<RESPONSE> {
        this.lang = lang
        return this
    }

    private var code: String? = null
    fun code(code: String?): ListResponseWrapper<RESPONSE> {
        this.code = code
        return this
    }

    private var message: String? = null
    fun message(message: String?): ListResponseWrapper<RESPONSE> {
        this.message = message
        return this
    }

    private var data: List<RESPONSE>? = null
    fun data(data: List<RESPONSE>?): ListResponseWrapper<RESPONSE> {
        this.data = data
        return this
    }

    companion object {
        fun <RESPONSE0> create(): ListResponseWrapper<RESPONSE0> {
            return ListResponseWrapper()
        }
    }
}
